webhook/plugin.py

```diff
@@ -0,0 +1,14 @@
+from typing import Any
+
+import httpx
+
+
+async def webhook_request(url: str, body: Any) -> Any:
```

Comment:
Is there a reason why the body's type and return type is `Any`?  I think the type checker will not catch any issues here and that seems like an issue


```suggestion
async def webhook_request(url: str, body: dict[str, Any]) -> dict[str, Any]:
```

Reply:
In general, `body` and the return value are arbitrary data structures that can be encoded to and decoded from JSON, respectively.  There are enough trade-offs around what that type should even be that Python does not provide one in the type hints of the standard library.  See the discussion in [this issue](https://github.com/python/typing/issues/182).  [The final comment](https://github.com/python/typing/issues/182#issuecomment-1320974824) summarizes nuances around type-checking.  Given the nuance, particularly around invariance and mutability, and the generality of webhook calling, it's not obvious to me what trade-off to make.

Similarly, the `json` parameter and return value of `.json()` are `Any`, presumably for the same reasons.

Furthermore, plugins do not yet support unions, as unions are generally very prone to introduce ambiguity in deserialization, and there is no mechanism to disambiguate.  [The one that exists for built-in actions](https://github.com/d8e-ai/alda/blob/0a7db08e381298d672c06c7f48f06e6c2626006f/v3/common.py#L223-L232) is pretty fragile and requires some relatively careful thinking, so I am resisting adding support to plugins until we have a very very good reason.

Do you have an example of a problem case, as I think a worked example would be useful to explore these trade-offs.